AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthConfirmSignInTask.swift

```diff
@@ -68,19 +73,11 @@ class AWSAuthConfirmSignInTask: AuthConfirmSignInTask {
                    } else if case .resolvingChallenge(let challengeState, _, _) = signInState {
                        switch challengeState {
                        case .waitingForAnswer:
-                           // Convert the attributes to [String: String]
-                           let attributePrefix = AuthPluginConstants.cognitoIdentityUserUserAttributePrefix
-                           let attributes = pluginOptions?.userAttributes?.reduce(
-                               into: [String: String]()) {
-                                   $0[attributePrefix + $1.key.rawValue] = $1.value
-                               } ?? [:]
-                           let confirmSignInData = ConfirmSignInEventData(
-                               answer: self.request.challengeResponse,
-                               attributes: attributes,
-                               metadata: pluginOptions?.metadata)
-                           let event = SignInChallengeEvent(
-                               eventType: .verifyChallengeAnswer(confirmSignInData))
-                           await authStateMachine.send(event)
+                           guard let result = try UserPoolSignInHelper.checkNextStep(signInState) else {
+                               continue
```

Comment:
Do you think a labeled statement (maybe `StateSequenceLoop`) could help here to clearly see the loop's control flow points.